CORE-44969 Changing setCustomerIds command name and options.

Aaronius opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Old API:

```
alloy("setCustomerIds", {
  // IDs here
});
```

New API:

```
alloy("syncIdentity", {
  identities: {
    // IDs here
  }
});
```
<!--- Describe your changes in detail -->

## Related Issue
https://jira.corp.adobe.com/browse/CORE-44969
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context
Request by PM as well as flexibility to add options later.
<!--- Why is this change required? What problem does it solve? -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

Aaronius commented on src/components/Identity/userIds/createUserIds.js:
```diff
@@ -69,5 +69,5 @@ export default ({
   };
 
-  return customerIds;
+  return userIds;
```
I personally find that having this module named `customerIds`/`userIds` is confusing, but I didn't want to get caught up in debates so I didn't attempt to change the pattern for this PR.

jonsnyder commented on src/components/Identity/createComponent.js:
```diff
@@ -36,9 +36,9 @@ export default ({
+      syncIdentity: {
         run: options => {
-          return customerIds.sync(options);
+          return userIds.sync(options.userIds);
```
We should add some validation so we don't get an error if the user passes the wrong key.
```
objectOf({userIds: objectOf({}).required()}).noUnknownFields().required()
```

Aaronius commented:
I've added an options validator along with unit tests. We probably should discuss whether/how we'd like to test options validators in the e2e tests across all commands. I've added a JIRA ticket for making sure we have validation set up for all commands and to try to standardize our testing approach: https://jira.corp.adobe.com/browse/CORE-44974

It sounds like since the time this PR was created, there's some hesitancy from PM around whether we'll be using `userIds` or `customerIds`. I'll keep this open a little longer until we either need to merge or receive final word.

Aaronius commented:
As of a few minutes ago, this:

```
alloy("syncIdentity", {
  userIds: {
    // IDs here
  }
});
```

will be changing to this:

```
alloy("syncIdentity", {
  identities: {
    // IDs here
  }
});
```

I'll try to make the changes tonight.
